app/Providers/AppServiceProvider.php

```diff
@@ -119,6 +121,28 @@ public function boot()
             return $this;
         });
 
+        Mail::extend('brevo', function () {
+            return (new BrevoTransportFactory)->create(
+                new Dsn(
+                    'brevo+api',
+                    'default',
+                    config('services.brevo.key')
+                )
+            );
+        });
+        // Mailer::macro('brevo_config', function (string $key) {
+        //     // @phpstan-ignore /** @phpstan-ignore-next-line **/
+        //     Mail::setSymfonyTransport((new BrevoTransportFactory)->create(
+        //         new Dsn(
+        //             'brevo+api',
+        //             'default',
+        //             $key
+        //         )
+        //     ));
+
+        //     return $this;
+        // });
```

Comment:
@turbo124 can you please test this behavior yourself. I was not able to do so :(

Reply:
we'll want to use the macro _definitely_ here extend:: is not appropriate for our particular use case as the brevo configuration is set at runtime

Reply:
I think empty() would be best method to use to check if the configuration has been set for the api key

Reply:
done